### ISSUE TITLE
nuitka MacOS Compiled app is looking for image resources on my desktop instead of in the "*.app"

### ISSUE THREAD

**Issue by isovsisland** [Nuitka/Nuitka#3213]
- Please check whether the bug was already reported or fixed.
**User Response:** Checked and posted issue on Discord. Kay Hayen respond to post the issue on GitHub.

- Please check whether the issue appears when you run the code with Python, if it does, it's not a
  Nuitka issue.
**User Response:** It does not. The code runs fine as a python script using PyCharm.

- Please check out if the develop version of Nuitka works better for you.
**User Response:** I will give it a try, but following instructions provided by Ms. Hayen

  Download source, packages [from here](https://nuitka.net/doc/download.html) where you will also
  find instructions how to do it via pip or git.
**User Response:** I have the instructions and researched various sources on the internet with no luck for a solution.

- Nuitka version, full Python version, flavor, OS, etc. as output by *this exact* command.
  > python -m nuitka --version

**User Response:** Command results below.
python3 -m nuitka --version
2.5.1
Commercial: None
Python: 3.12.1 (v3.12.1:2305ca5144, Dec  7 2023, 17:23:38) [Clang 13.0.0 (clang-1300.0.29.30)]
Flavor: CPython Official
Executable: /Library/Frameworks/Python.framework/Versions/3.12/bin/python3
OS: Darwin
Arch: arm64
macOSRelease: 14.6.1
Version C compiler: /usr/bin/clang (clang 16.0.0).

- How did you install Nuitka and Python
**User Response:** Python download dmg file with installer. nuitka installed via pip. virtualenv not used.

- The specific PyPI names and versions
**User Response:**
pip3 --version
pip 24.3.1 from /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pip (python 3.12)

  It should be taken from this output if there specific packages involved, e.g. `numpy`, you are
  expected to shorten this to the relevant ones.
  > python -m pip list -v
User Response: My project only requires tkinter, but below is a complete list of the pip modules from my machine.

python3 -m pip list -v
Package     Version Location                                                                        Installer
----------- ------- ------------------------------------------------------------------------------- ---------
altgraph    0.17.4  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
macholib    1.16.3  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
modulegraph 0.19.6  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
Nuitka      2.5.1   /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
ordered-set 4.1.0   /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
packaging   24.2    /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
pillow      11.0.0  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
pip         24.3.1  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
py2app      0.28.8  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
setuptools  70.3.0  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip
zstandard   0.23.0  /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages pip

- Also supply a Short, Self Contained, Correct, Example
**User Response:** Code example below. The compiled code does not provide an error window. There is a "blip" on the screen indicating that something happened but no window appears. If the code is launched via terminal window it responds that it cannot find the image file "/Users/user_name/Desktop/rsrc/psf_image.png" if the "rsrc" folder is placed on the desktop, the app runs fine from the terminal but still does not run when the file is launched from Finder. I have tried compiling with both include-file and include-dirs options with the same result. In review of the app package contents the MacOS folder shows the image file indicating that nuitka recognized the command and copied it to the app.
~~~
"""
111724
Python 3.12.1 & tkinter

Python script compiled with nuitka is looking for app resources on desktop instead of in *.app

"""
import tkinter as tk
from tkinter.constants import *
from PIL import Image, ImageTk

TITLE = "nuitka MacOS issue with Compiled Python Script"
IMAGE = './rsrc/psf_image.png'

class Main(tk.Tk):
    def __init__(self):
        super().__init__()
        self.title(TITLE)

        open_img = Image.open(IMAGE)
        img = ImageTk.PhotoImage(open_img)

        label_img = tk.Label(self, text="Image", image=img)
        label_img.pack()

        self.mainloop()

if __name__ == '__main__':
    Main()
~~~

**User Response:** nuitka Options used.
python3 -m nuitka --standalone --enable-plugin=tk-inter --macos-create-app-bundle nuitka_macos_app_issue.py

**Comment by kayhayen** [comment]
You cannot use relative filenames in your program and expect it work, check out: https://nuitka.net/user-documentation/common-issue-solutions.html#standalone-finding-files

**Comment by isovsisland** [comment]
The problem occurs with or without relative filenames. Please note that this problems does not occur with other compilers such as a py2app or pyinstaller.
